eloquentjs/20_1_search_tool/src/nodeGREP.ts

```diff
@@ -0,0 +1,54 @@
+import * as  fs from "fs"
+
+
+let regeExpString = process.argv[2];
+let filesAndFolders = process.argv.slice(3);
+
+if (!regeExpString || filesAndFolders.length === 0) {
+    throw "Missing arguments"
+}
+
+console.log(nodeGREP(regeExpString, filesAndFolders));
+
+/**
+ * Searches provided files, folders, and subfolders for file content that matches regexp
+ * Returns list of matching files
+ * @param regexpString 
+ * @param searchTargets 
+ * @returns 
+ */
+function nodeGREP(regexpString: string, searchTargets: Array<string>): Array<string> {
+    let validSearchTargets: Array<string> = searchTargets.filter(target => fs.existsSync(target));
+    let invalidSearchTargets: Array<string> = searchTargets.filter(target => !fs.existsSync(target));
+
+    for (let target of invalidSearchTargets) {
+        console.log(`Target does not exist: ${target}`)
+    }
+
+    const regexp = new RegExp(regexpString);
+    const matchingFiles: Array<string> = [];
```

Comment:
Visually associating this with the inner recursive function makes the code more readable imo